src/osd/PGLog.cc

```diff
@@ -272,7 +272,7 @@ void PGLog::proc_replica_log(
   // We need to preserve the original crt before it gets updated in rewind_from_head().
   // Later, in merge_object_divergent_entries(), we use it to check whether we can rollback
   // a divergent entry or not.
-  eversion_t original_crt = log.get_can_rollback_to();
+  eversion_t original_crt = olog.get_can_rollback_to();
```

Comment:
Can you please explain why this is required? 
The comment above states why we want to preserve the original log's crt. The replica log's(olog) crt gets passed to _merge_divergent_entries() and _merge_object_divergent_entries() to check rollbackability without getting modified, so why do we need to preserve it?

Reply:
@neha-ojha  sorry for not explain this.  
I agree that we do not need to preserve the replica log's crt, but this is not the key point...
the key point here I think is we pass the primary log's crt to the _merge_divergent_entries(), in other words, _merge_divergent_entries() will merge log according to primary log's crt. I think this is wrong place.
When we are merging the replica log, we should rollback according to replica log's crt.

Reply:
here we could simply pass the olog.get_can_rollback_to() in the place of original_crt

Reply:
The title is not appropriate...

Reply:
@Aran85 This makes much more sense now, thanks for the explanation! Could you please add a comment with this explanation for better understanding of the code in future.